api/cluster/controller.go

```diff
@@ -156,6 +158,13 @@ func (k *controller) Deploy(modelService *models.Service) (*models.Service, erro
 
 	s, err = k.waitInferenceServiceReady(s)
 	if err != nil {
+		// remove newly created inferenceservice when got error
```

Comment:
This is unrelated to queue implementation right? it's for the fix when deployment is considered fail but we didn;t garbage collect the resource.

Reply:
yes, it is not related to queue implementation. I changed this also because it is related to model deployment